analyses/upstream-analysis/04_run_filter_object.Rmd

```diff
@@ -0,0 +1,381 @@
+print(DefaultAssay(seurat_obj) <- "RNA") # Switch the default to RNA
+saveRDS(seurat_obj, file = paste0(results_dir, "/", "seurat_obj_merged.rds")) # Save merged object
+#############################################################################
+# Estimate `cells_number_seurat_qc` per sample_name
+cells_number_seurat_qc <- table(seurat_obj@meta.data$orig.ident) %>%
+    as.data.frame() %>% 
+    mutate(sample_name = Var1,
+           cells_number_seurat_qc = Freq) %>% 
+    dplyr::select(-Var1, -Freq)
+```
+
+# Filter ambient RNA
+Next, we will use the corrected matrices from all samples as generated by [SoupX](https://cran.r-project.org/web/packages/SoupX/vignettes/pbmcTutorial.html). This is an optional step, so it will be skipped or not as defined in `params`. 
+
+```{r filter-ambient-RNA, echo=TRUE}
+# Indicates whether or not to use the SoupX corrected matrix for filtering or to skip it
+if (use_SoupX_filtering == "YES"){
+  print_message <- print("we will use SoupX corrected matrix")
+  SoupX_seurat_obj <- readRDS(SoupX_file) # Read SoupX_seurat_obj
+  colnames(seurat_obj)[which(!(colnames(seurat_obj) %in% colnames(SoupX_seurat_obj) ))]
+  seurat_obj <- subset(seurat_obj, cells=colnames(seurat_obj)[which(!(colnames(seurat_obj) %in% colnames(SoupX_seurat_obj) ))], invert=TRUE)
+  SoupX_seurat_obj <- SoupX_seurat_obj[,colnames(seurat_obj)]
+  seurat_obj[["RNA_SoupX"]] <- CreateAssayObject(counts = SoupX_seurat_obj)
+  DefaultAssay(seurat_obj) <- "RNA_SoupX"
+  cat("Cells_number_after_filter_ambient_RNA per orig.ident", "\n")
+  print(table(seurat_obj@meta.data$orig.ident))
+  cat("Cells_number_after_filter_ambient_RNA per ID", "\n")
+  print(table(seurat_obj@meta.data$ID))
+  } else {
+    print_message <- print("we will skip usage of the SoupX corrected matrix")}
+```
+
+Here, `r print_message`.
+
+# Filter doublets out
+Next, we will remove any doublets from all samples as identified by [ScDblFinder](https://bioconductor.org/packages/devel/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html).
+
+```{r filter-doublets, echo=TRUE}
+# Read scDblFinder results
+scDblFinder_obj <- readRDS(scDblFinder_file)
+
+#############################################################################
+# Estimate `cells_number_raw` per sample_name
+cells_number_raw <- table(scDblFinder_obj@meta.data$orig.ident) %>%
+    as.data.frame() %>% 
+    mutate(sample_name = Var1,
+           cells_number_raw = Freq) %>% 
+    dplyr::select(-Var1, -Freq)
+
+#############################################################################
+# Add metadata
+seurat_obj <- AddMetaData(seurat_obj, scDblFinder_obj@meta.data)
+
+##########################################################################################################
+# Report Singlets 
+singlet_cells <- colnames(scDblFinder_obj)[which(scDblFinder_obj@meta.data$scDblFinder.class == "singlet")]
+seurat_obj_doublets <- subset(seurat_obj, cells = singlet_cells, invert = TRUE)
```

Comment:
I feel like it may be slightly confusing to have this command and the command below under the section commented as reporting singlets, since these two commands are actually saving information about the cells that are label as doublets.

Reply:
This is now addressed here 0d107fdf0c06efb24a0cdd32fda4e191fa88eff6